data_preparation/global_AgERA5/download_agera5.py

```
@@ -0,0 +1,123 @@
+    "Minimum_Temperature": ("2m_temperature", "24_hour_minimum"),
+    "Mean_Temperature": ("2m_temperature", "24_hour_mean"),
+    "Solar_Radiation_Flux": ("solar_radiation_flux", None),
+    "Precipitation_Flux": ("precipitation_flux", None),
+}
+
+
+# @author: Abdelrahman Saleh
+def get_agera5_params(sel_param, start_year, end_year, area=None):
+    if sel_param in AgERA5_params:
+        var, stat = AgERA5_params[sel_param]
+
+        # Construct the return dictionary
+        retrieve_params = {
+            "version": "1_1",
+            "format": "zip",
+            "variable": var,
+            "year": [str(year) for year in range(start_year, end_year + 1)],
+            "month": [
+                "01",
+                "02",
+                "03",
+                "04",
+                "05",
+                "06",
+                "07",
+                "08",
+                "09",
+                "10",
+                "11",
+                "12",
+            ],
+            "day": [
+                "01",
+                "02",
+                "03",
+                "04",
+                "05",
+                "06",
+                "07",
+                "08",
+                "09",
+                "10",
+                "11",
+                "12",
+                "13",
+                "14",
+                "15",
+                "16",
+                "17",
+                "18",
+                "19",
+                "20",
+                "21",
+                "22",
+                "23",
+                "24",
+                "25",
+                "26",
+                "27",
+                "28",
+                "29",
+                "30",
+                "31",
+            ],
+        }
+
+        if stat is not None:
+            retrieve_params["statistic"] = stat
+
+        if area is not None:
+            retrieve_params["area"] = area
+
+        return retrieve_params
+    else:
+        return None
```

We can return an assertion error if the user puts a parameter outside the defined parameters in AgERA5_params.